src/database/schema/ChatSchema.ts

```diff
@@ -174,6 +175,28 @@ class ChatSchema implements BaseDbSchema {
     return Promise.resolve(this.fromDatabaseObject(rows.raw()[0]));
   }
 
+  static async getPendingMessages(db: SQLiteDatabase): Promise<ChatSchema[]> {
+    const selectQuery = `SELECT A.*,
+      B.username, 
+      B.country_code, 
+      B.created_at as user_created_at, 
+      B.updated_at as user_updated_at, 
+      B.last_active_at, 
+      B.profile_picture, 
+      B.bio, 
+      B.is_banned,
+      C.channel_type
+      FROM ${ChatSchema.getTableName()} A
+      INNER JOIN ${UserSchema.getTableName()} B
+      ON A.user_id = B.user_id
+      INNER JOIN ${ChannelListSchema.getTableName()} C
+      ON A.channel_id = C.id
+      WHERE A.status = 'pending';`;
+
+    const [{rows}] = await db.executeSql(selectQuery, ['pending Messages']);
```

Comment:
`['pending Messages']` is not important.
can be deleted

Reply:
oh wait mas checking, tadi soalnya dia minta id waktu aku hapus

Reply:
oiya aman ternyata, revised mas @fajarism